src/bin/gpfusion/gpbridgeapi.c

```diff
@@ -273,6 +277,26 @@ void gpbridge_export_start(PG_FUNCTION_ARGS)
 
 }
 
+void parse_namespace(char *uri, char** extracted_namespace){
+	int i = 0;
+	int len = strlen(uri);
+
+	while(uri[i] != '*' && i < len){
```

Comment:
Actually, shouldn't we parse the URI as part of the parse function we already have? that way we can know that we parsed everything else before we expect to read the delimiter and the dfs_address - similar to the way it's implemented right now, when we parse dfs_address that was appended by hd_work_mgr.